task_processing/interfaces/runner.py

```diff
@@ -14,9 +16,12 @@ class Runner(object):
     """
 
     @abc.abstractmethod
-    def run(self, task_config):
+    def run(self, task_config, task_id=None):
         pass
 
     @abc.abstractmethod
     def kill(self, task_id):
         pass
+
+    def new_task_id(self):
```

Comment:
I don't think this should be part of the interface. base62_random() is util method that people can just import and use.